pkg/importer/dataStream.go

```diff
@@ -59,22 +60,37 @@ var qemuOperations = image.NewQEMUOperations()
 
 // DataStream implements the ReadCloser interface
 type DataStream struct {
-	url         *url.URL
-	Readers     []reader
-	buf         []byte // holds file headers
-	qemu        bool
-	Size        int64
-	accessKeyID string
-	secretKey   string
-	ctx         context.Context
-	cancel      context.CancelFunc
+	*DataStreamOptions
+	url     *url.URL
+	Readers []reader
+	buf     []byte // holds file headers
+	qemu    bool
+	Size    int64
+	ctx     context.Context
+	cancel  context.CancelFunc
 }
 
 type reader struct {
 	rdrType int
 	rdr     io.ReadCloser
 }
 
+// DataStreamOptions contains all the values needed for importing from a DataStream.
+type DataStreamOptions struct {
+	// Dest is the destination path of the contents of the stream.
+	Dest string
+	// Endpoint is the endpoint to get the data from for various Sources.
+	Endpoint string
+	// AccessKey is the access key for the endpoint, can be blank.
```

Comment:
worth mentioning here that it should be encoded in base64? or is it enough to rely on the readme?

Reply:
@danielerez makes a good point, it probably would be good to add the b64 encoding requirement to the godoc strings.

Reply:
Sure, done.